drivers/ethernet/eth_stm32_hal.c

```diff
@@ -152,25 +199,65 @@ static int eth_tx(struct device *dev, struct net_pkt *pkt)
 		goto error;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	const uint32_t cur_tx_desc_idx = 0;  /* heth->TxDescList.CurTxDesc; */
+#endif
+
 	dma_tx_desc = GET_FIRST_DMA_TX_DESC(heth);
 	while (IS_ETH_DMATXDESC_OWN(dma_tx_desc) != (uint32_t)RESET) {
 		k_yield();
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	dma_buffer = dma_tx_buffer[cur_tx_desc_idx];
+#else
 	dma_buffer = (uint8_t *)(dma_tx_desc->Buffer1Addr);
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
 
 	if (net_pkt_read(pkt, dma_buffer, total_len)) {
 		res = -EIO;
 		goto error;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	ETH_BufferTypeDef tx_buffer_def[ETH_TXBUFNB];
+
+	memset(tx_buffer_def, 0, ETH_TXBUFNB*sizeof(ETH_BufferTypeDef));
+
+	tx_buffer_def[cur_tx_desc_idx].buffer = dma_buffer;
+	tx_buffer_def[cur_tx_desc_idx].len = total_len;
+	tx_buffer_def[cur_tx_desc_idx].next = NULL;
+
+	tx_config.Length = total_len;
+	tx_config.TxBuffer = tx_buffer_def;
+
+	hal_ret = HAL_ETH_Transmit(heth, &tx_config, ETH_DMA_TRANSMIT_TIMEOUT);
+#else
 	hal_ret = HAL_ETH_TransmitFrame(heth, total_len);
+#endif /* CONFIG_SOC_SERIES_STM32H7X */
+
 	if (hal_ret != HAL_OK) {
-		LOG_ERR("HAL_ETH_TransmitFrame failed");
+		LOG_ERR("HAL_ETH_Transmit{Frame}: failed!");
 		res = -EIO;
 		goto error;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
+	/* When Transmit Buffer Unavailable flag is set, clear it and issue a
+	 * Resume processing the Transmit descriptors.
+	 */
+	if ((heth->Instance->DMACSR & ETH_DMACSR_TBU) != (uint32_t)RESET) {
+		/* @TODO: Must be done or skipped! */
```

Comment:
Rather than merging a commented section, I'd prefer this whole piece of code being left as a follow up PR with a tag "Block: HW Tests"

Reply:
Will be deleted within next push.